ktlint_code_style = android not working

<!-- The bug you're experiencing might have already be reported!
Please search in the [issues](https://github.com/pinterest/ktlint/issues) before creating one. -->

## Expected Behavior
<!---Tell us what should happen. -->
<!-- If relevant, provide a link to the relevant section in the [Kotlin Coding Conventions](https://kotlinlang.org/docs/coding-conventions.html) or [Android Kotlin Style Guide](https://developer.android.com/kotlin/style-guide).  -->

<!-- Note: Ktlint should not produce any code which conflict with IntelliJ default code formatting. So please verify that your expectation is accepted by IntelliJ default code formatting. -->

ktlint_code_style should be applied. but when run gradle task of ktlint, exception occurred

## Observed Behavior
<!---Tell us what happens instead of the expected behavior -->
<!--- Provide the exact command which was executed but please -->
<!--- ensure that the flag "--verbose" is added to the -->
<!--- command as well. Provide the output of this command. -->

ClassCastException occurred

when I see the document in https://pinterest.github.io/ktlint/rules/configuration/
we can configure the code style to android. but the property is not working...

```
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.ClassCastException: class java.lang.String cannot be cast to class com.pinterest.ktlint.core.api.DefaultEditorConfigProperties$CodeStyleValue (java.lang.String is in module java.base of loader 'bootstrap'; com.pinterest.ktlint.core.api.DefaultEditorConfigProperties$CodeStyleValue is in unnamed module of loader 'app')
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at com.pinterest.ktlint.internal.KtlintCommandLine.parallel(KtlintCommandLine.kt:578)
	at com.pinterest.ktlint.internal.KtlintCommandLine.parallel$default(KtlintCommandLine.kt:536)
	at com.pinterest.ktlint.internal.KtlintCommandLine.lintFiles(KtlintCommandLine.kt:283)
	at com.pinterest.ktlint.internal.KtlintCommandLine.run(KtlintCommandLine.kt:235)
	at com.pinterest.ktlint.Main.main(Main.kt:31)
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class com.pinterest.ktlint.core.api.DefaultEditorConfigProperties$CodeStyleValue (java.lang.String is in module java.base of loader 'bootstrap'; com.pinterest.ktlint.core.api.DefaultEditorConfigProperties$CodeStyleValue is in unnamed module of loader 'app')
	at com.pinterest.ktlint.core.api.UsesEditorConfigProperties$DefaultImpls.getEditorConfigValue(UsesEditorConfigProperties.kt:50)
	at com.pinterest.ktlint.core.internal.VisitorProvider.getEditorConfigValue(VisitorProvider.kt:17)
	at com.pinterest.ktlint.core.internal.VisitorProvider.isNotDisabled(VisitorProvider.kt:132)
	at com.pinterest.ktlint.core.internal.VisitorProvider.visitor$ktlint_core(VisitorProvider.kt:43)
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class com.pinterest.ktlint.core.api.DefaultEditorConfigProperties$CodeStyleValue (java.lang.String is in module java.base of loader 'bootstrap'; com.pinterest.ktlint.core.api.DefaultEditorConfigProperties$CodeStyleValue is in unnamed module of loader 'app')

	at com.pinterest.ktlint.core.internal.VisitorProvider.visitor$ktlint_core$default(VisitorProvider.kt:37)
	at com.pinterest.ktlint.core.KtLint.lint(KtLint.kt:140)
	at com.pinterest.ktlint.internal.FileUtilsKt.lintFile(FileUtils.kt:174)
	at com.pinterest.ktlint.internal.KtlintCommandLine.process(KtlintCommandLine.kt:384)
	at com.pinterest.ktlint.internal.KtlintCommandLine.access$process(KtlintCommandLine.kt:48)
	at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$3.invoke$lambda-1(KtlintCommandLine.kt:274)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

## Steps to Reproduce
<!--- Provide a code example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Minimize the example to the bare -->
<!--- minimum required to reproduce the bug. Small, well written -->
<!--- example are proven to be resolved quicker. -->

add "ktlint_code_style = official" to project or module level .editorconfig

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version of ktlint used: 0.46.1
* Relevant parts of the `.editorconfig` settings "ktlint_code_style"
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): gradle plugin ( android )
* Version of Gradle used (if applicable):
* Operating System and version: macOs 11.6.2

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -126,7 +126,7 @@ The callback function provided as parameter to the format function is now called
 * When a glob is specified then ensure that it matches files in the current directory and not only in subdirectories of the current directory ([#1533](https://github.com/pinterest/ktlint/issue/1533)).
 * Execute `ktlint` cli on default kotlin extensions only when an (existing) path to a directory is given. ([#917](https://github.com/pinterest/ktlint/issue/917)).
 * Invoke callback on `format` function for all errors including errors that are autocorrected ([#1491](https://github.com/pinterest/ktlint/issues/1491))
-
+* Prevent class cast exception on ".editorconfig" property `ktlint_code_style`  ([#1559](https://github.com/pinterest/ktlint/issues/1559))
 
 ### Changed
 

--- a/ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/api/UsesEditorConfigProperties.kt
+++ b/ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/api/UsesEditorConfigProperties.kt
@@ -47,9 +47,23 @@ public interface UsesEditorConfigProperties {
         require(editorConfigProperties.contains(editorConfigProperty)) {
             "EditorConfigProperty '${editorConfigProperty.type.name}' may only be retrieved when it is registered in the editorConfigProperties."
         }
-        val codeStyle = getEditorConfigValue(codeStyleSetProperty, official)
-        return getEditorConfigValue(editorConfigProperty, codeStyle)
+
+        return getEditorConfigValue(editorConfigProperty, getEditorConfigCodeStyle())
     }
+
+    /**
+     * The code style property does not need to be defined in the [editorConfigProperties] of the class that defines
+     * this interface. Those classed should not need to be aware of the different coding styles except when setting
+     * different default values. As the property is not defined in the [editorConfigProperties] the value needs to
+     * be parsed explicitly to prevent class cast exceptions.
+     */
+    private fun EditorConfigProperties.getEditorConfigCodeStyle() =
+        codeStyleSetProperty
+            .type
+            .parse(
+                get(codeStyleSetProperty.type.name)?.sourceValue
+            ).parsed
+            ?: official
 
     /**
      * Get the value of [EditorConfigProperty] based on loaded [EditorConfigProperties] content for the current
@@ -61,8 +75,10 @@ public interface UsesEditorConfigProperties {
             "EditorConfigProperty '${editorConfigProperty.type.name}' may only be retrieved when it is registered in the editorConfigProperties."
         }
         val editorConfigPropertyValues = getUserData(KtLint.EDITOR_CONFIG_PROPERTIES_USER_DATA_KEY)!!
-        val codeStyle = editorConfigPropertyValues.getEditorConfigValue(codeStyleSetProperty, official)
-        return editorConfigPropertyValues.getEditorConfigValue(editorConfigProperty, codeStyle)
+        return editorConfigPropertyValues.getEditorConfigValue(
+            editorConfigProperty,
+            editorConfigPropertyValues.getEditorConfigCodeStyle()
+        )
     }
 
     private fun <T> EditorConfigProperties.getEditorConfigValue(

